### PR TITLE
Added AST based LANG translation string detection

### PR DESCRIPTION
**Pull request by A-UNDERSCORE-D**
This could probably benefit from:
- deduping (at least for some output options)
- the ability to direct compare with the existing template file

### CODE REVIEW

File: scripts/find_localised_strings.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        return get_func_name(thing.value)
+
+    else:
+        return ""
+
+
+def get_arg(call: ast.Call) -> str:
+    """Extract the argument string to the translate function."""
+    if len(call.args) > 1:
+        print("??? > 1 args", call.args)
```
Comment (A-UNDERSCORE-D) [inline scripts/find_localised_strings.py:23]: STDERR this or remove it.

**Comment by A-UNDERSCORE-D** [comment]
okay the only thing this now needs is a update to the general codebase with LANG comments

File: scripts/find_localised_strings.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        return out[1]
+    elif out[0] is not None:
+        return out[0]
+
+    return None
+
+
+def scan_file(path: pathlib.Path) -> list[ast.Call]:
+    """Scan a file for ast.Calls."""
+    data = path.read_text()
```
Comment (Athanasius) [inline scripts/find_localised_strings.py:91]: `path.read_text(encoding='utf-8')` else it crashes out on hotkey.py in the `DISPLAY` dict.

**Comment by Athanasius** [comment]
```python
scripts/find_localised_strings.py:8:1: I001 isort found an import in the wrong position
scripts/find_localised_strings.py:10:1: F401 'io' imported but unused
scripts/find_localised_strings.py:10:1: I001 isort found an import in the wrong position
scripts/find_localised_strings.py:54:1: D103 Missing docstring in public function
scripts/find_localised_strings.py:134:1: D103 Missing docstring in public function
scripts/find_localised_strings.py:148:1: D101 Missing docstring in public class
scripts/find_localised_strings.py:156:1: D102 Missing docstring in public method
scripts/find_localised_strings.py:161:1: D101 Missing docstring in public class
scripts/find_localised_strings.py:166:1: D102 Missing docstring in public method
scripts/find_localised_strings.py:177:1: CCR001 Cognitive complexity is too high (20 > 15)
```

Other than that, and the read_text encoding tweak, this looks very good.  If we can get it merged then I'll work on adding comments today.

**Pull request merged**